### ISSUE TITLE
Adding github storage provider

### ISSUE THREAD

**Issue by chrisUsick** [pkgcloud/pkgcloud#427]
Just checking with the community to see if there is any reason why it wouldn't be useful to create a github storage provider, or it hasn't been done already?

**Comment by kenperkins** [comment]
Totally. Can you point to an API for it?

**Comment by chrisUsick** [comment]
[Github official documentation is here](https://developer.github.com/v3/repos/contents/).   
Here are 2 competing node APIs to github api and they both look quite mature.
- [octonode](https://github.com/pksunkara/octonode)  is edited more recently, has more forks
- [node-github](https://github.com/ajaxorg/node-github)

**Comment by kenperkins** [comment]
so you're recommending using an existing library similar to how we do with AWS?

**Comment by chrisUsick** [comment]
Yes. I saw that the AWS and Google providers use existing libraries. I would recommend investigating each library and seeing if it would be more work to integrate them into the pkgcloud system or just build from scratch, using the the github API directly. What do you think would be less work?

**Comment by kenperkins** [comment]
Certainly be less work to use an existing package; I don't like the dependency bloat tho. We've talked about breaking pkgcloud into pieces, i.e. `pkgcloud-storage`, etc, so it would have narrower dependencies.

**Comment by chrisUsick** [comment]
Okay. Yes, dependency bloat is a large problem with nodejs I've found. It can start to bring down the original snappiness that nodejs offers the way npm manage dependencies. I've worked with a framework called [loopback](http://loopback.io) (which uses pkgcloud) and it has tremendous dependencies. It only uses the storage functionality of pkgcloud so it would probably benefit from having pkgcloud split up.

**Comment by dantman** [comment]
@kenperkins If you're worried about dependency bloat wouldn't pkgcloud + pkgcloud-{amazon,rackspace,openstack,etc...} make more sense?

**Comment by chrisUsick** [comment]
Yeah, that would make more sense.
